### PR TITLE
Update GenAI NPU doc

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by kblaszczak-intel — APPROVED** [review]

**Review by msmykx-intel — APPROVED** [review]

File: docs/articles_en/learn-openvino/llm_inference_guide/genai-guide-npu.rst [hunk 1]
```diff
@@ -34,7 +34,7 @@ Run generation using OpenVINO GenAI
 
       python -m venv run-npu-env
       run-npu-env\Scripts\activate
-      pip install openvino>=2024.3.1 openvino-tokenizers>=2024.3.1 openvino-genai>=2024.3.1
+      pip install --pre openvino==2024.3.0.dev20240807 openvino-tokenizers==2024.3.0.0.dev20240807 openvino-genai==2024.3.0.0.dev20240807 --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/pre-release
```
Comment (ilya-lavrenov) [inline docs/articles_en/learn-openvino/llm_inference_guide/genai-guide-npu.rst:37]: it's allowed to provide only release versions

Comment (TolyaTalamanov) [reply docs/articles_en/learn-openvino/llm_inference_guide/genai-guide-npu.rst:37]: The latest available version that contains NPU changes is `2024.3.0.dev20240807`. There is no `2024.3.1` unfortunately

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Comment by ilya-lavrenov** [comment]
I suppose we need to re-target to 2024.3 ? 
Because in upcoming 2024.4 we don't need custom version for NPU

**Comment by kblaszczak-intel** [comment]
If this is not going to be necessary for the 2024.4 release, I suggest we close this PR and in a separate one remove the article in the master branch.
@ilya-lavrenov What do you think?

**Comment by TolyaTalamanov** [comment]
> If this is not going to be necessary for the 2024.4 release, I suggest we close this PR and in a separate one remove the article in the master branch. In the 2024.3 version it is already implemented and published. @ilya-lavrenov What do you think?

The version of documentation that is published isn't correct, it refers to `2024.3.1` version that not existing, we need to replace it with the version proposed in this PR

**Review by kblaszczak-intel — APPROVED** [review]

**Pull request merged**